specs/core/0_beacon-chain.md

```diff
@@ -1691,7 +1644,7 @@ def process_ejections(state: BeaconState) -> None:
     """
     for index in active_validator_indices(state.validator_registry):
         if state.validator_balances[index] < EJECTION_BALANCE:
-            update_validator_status(state, index, new_status=EXITED_WITHOUT_PENALTY)
+            exit_validator(state, index)
```

Comment:
`exit_validator` checks `if validator.exit_slot < state.slot + ENTRY_EXIT_DELAY: return`. Will it work here since `validator.exit_slot` was set to `FAR_FUTURE_SLOT`

Reply:
`exit_validator`  does that check as a blocking condition so validators are not exited twice.

In the case that a validator is not exited then `exit_slot` is _not_ less than `state.slot + ENTRY_EXIT_DELAY` so it does not immediately return and instead performs the exit